scripts/get_sources_from_srpm

```diff
@@ -31,31 +32,23 @@ exit_get_sources() {
 }
 
 if [ "x$releasever" != "x" ]; then
-    if [[ ! "$releasever" =~ ^[1-9][0-9]$ ]]; then
-        errecho "Invalid Fedora release format"
+    if [[ ! "$releasever" =~ ^[1-9][0-9]$ ]] && [ "$releasever" != "rawhide" ]; then
+        errecho "Invalid release format"
         exit 1
     fi
 else
     errecho "Please provide Fedora release"
     exit 1
 fi
 
-release_url="https://dl.fedoraproject.org/pub/fedora/linux/releases/${releasever}/Everything/source/tree/Packages/${src_rpm::1}/${src_rpm}"
-update_url="https://dl.fedoraproject.org/pub/fedora/linux/updates/${releasever}/Everything/SRPMS/Packages/${src_rpm::1}/${src_rpm}"
-
 trap 'exit_get_sources' 0 1 2 3 6 15
 
 if [ "x$src_file" != "x" ] && [ "x$releasever" != "x" ]; then
     # download rpm
     tmpdir="$(mktemp -d -p "$localdir")"
-    if exist_url "$update_url"; then
-        wget -q "$update_url" -O "$tmpdir/$src_rpm.UNTRUSTED"
-    elif exist_url "$release_url"; then
-        wget -q "$release_url" -O "$tmpdir/$src_rpm.UNTRUSTED"
-    else
-        errecho "Cannot find source RPM: $src_rpm"
-        exit 1
-    fi
+    # shellcheck disable=SC2086
+    dnf -q download --destdir="$tmpdir" --source --releasever="$releasever" "$package"
+    mv "$tmpdir/$src_rpm" "$tmpdir/$src_rpm.UNTRUSTED"
```

Comment:
I think this will fail if for example dnf downloads newer package that named in `$src_rpm`.
But since `set -e` and automatic removal of tmpdir, it will fail on the safe side, so should be fine.